src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCommitter.cs

```diff
@@ -217,18 +229,24 @@ private void UpdateWorkspaceForResetOfTypedIdentifier(Workspace workspace, Solut
                 localUndoTransaction.AddUndo(undoPrimitiveBefore);
 
                 if (!workspace.TryApplyChanges(newSolution))
-                {
-                    Contract.Fail("Rename Tracking could not update solution.");
```

Comment:
fairly certain this would just corrupt state in the rename engine (due to the lack of `finally`s).